src/components/ProposalTable.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import {
+  Table,
+  TableHead,
+  ColumnHead,
+  TableBody,
+  TableRow,
+  RowHead,
+  RowCell,
+} from './Table';
+
+interface ProposalTableProps {
+  version: number;
+  values: {
+    fieldName: string,
+    position: number,
+    value: string,
+  }[],
+}
+
+export const ProposalTable = ({
+  values,
+  version,
+}: ProposalTableProps) => (
+  <Table>
+    <TableHead fixed>
+      <TableRow>
+        <ColumnHead>
+          Canonical Field
+        </ColumnHead>
+        <ColumnHead>
+          {`Version ${version}`}
+        </ColumnHead>
+      </TableRow>
+    </TableHead>
+    <TableBody>
+      {values.map(({ fieldName, position, value }) => (
+        <TableRow key={position}>
```

Comment:
Ah, is `position` actually unique in each dataset (e.g. safe to use for `key` on its own rather than as a string like `${fieldName}-${position}`)? I assumed it was relative to other fields with the same fieldName. I assume you're right, but just flagging it in case.

Reply:
Excellent question! I was slightly confident when I committed, but after doing some more reading and thinking, I am now more confident. (One of the better articles I've found was [React key attribute: best practices for performant lists](https://www.developerway.com/posts/react-key-attribute).)

I think the two key (hah) things to think about here is what state items in the list carry, when this list will change. 

Maintaining state across reordering or complete switches of underlying data is important to maintain correct behavior. The example from the article is instructive: if you have the first item in the list checked, and then change the sort, the checkmark should follow the item rather than the index. However, in our case, list items (`<TableRow>`s, here) do not carry any state, so I do not think there is a risk of incorrect behavior no matter what approach we take.

If correctness is not a concern, then that leaves performance. I believe this list will change under two circumstances:  when navigating from one proposal to another or toggling the version history. The version history is simpler: turning it on can only add rows at the end (fields that existed on previous versions of the proposal but not the current version of the proposal), and turning it off again will only delete rows at the end, so we could even use the `index` and be okay.

Switching proposals is more complicated, but reusing positions will allow React to maintain most of the existing DOM structure, changing only the values. Some positions might be missing, and others added (eg a proposal for an application form with field positions `[1, 2, 4]` switching to one with positions `[1, 2, 3, 5]`), and there'd be some performance loss from using `position` as a key rather than `index`, but on the other hand some navigations will be between proposals for the same application form, and in that case performance should be ideal - matching rows will match not only their `position` but also their row header.

Reply:
Really thorough and instructive explanation, thanks @jasonaowen!